src/preferences/dialog/dlgprefcolors.cpp

```diff
@@ -94,7 +118,84 @@ void DlgPrefColors::slotApply() {
-            ConfigKey("[Controls]", "auto_hotcue_colors"),
-            checkBoxAssignHotcueColors->isChecked());
+    int index = comboBoxHotcueDefaultColor->currentIndex();
+
+    if (index > 0) {
+        m_pConfig->setValue(ConfigKey("[Controls]", "auto_hotcue_colors"), false);
+        m_pConfig->setValue(ConfigKey("[Controls]", "HotcueDefaultColorIndex"), index - 1);
+    } else {
+        m_pConfig->setValue(ConfigKey("[Controls]", "auto_hotcue_colors"), true);
+        m_pConfig->setValue(ConfigKey("[Controls]", "HotcueDefaultColorIndex"), -1);
+    }
+}
+
+QPixmap DlgPrefColors::drawPalettePreview(const QString& paletteName) {
+    foreach (const ColorPalette& palette, mixxx::PredefinedColorPalettes::kPalettes) {
+        if (paletteName == palette.getName()) {
+            int count = math_max(palette.size(), 1);
+            int width = math_min((200 / count), 16);
+            QPixmap pixmap(count * width, 16);
+            pixmap.fill(Qt::black);
+            QPainter painter(&pixmap);
+            for (int i = 0; i < palette.size(); ++i) {
+                painter.setPen(mixxx::RgbColor::toQColor(palette.at(i)));
+                painter.setBrush(mixxx::RgbColor::toQColor(palette.at(i)));
+                painter.drawRect(i * width, 0, width, 16);
+            }
+            return pixmap;
+        }
+    }
+    return QPixmap();
+}
+
+void DlgPrefColors::slotTrackPaletteChanged(const QString& paletteName) {
+    QPixmap pixmap = drawPalettePreview(paletteName);
+    labelTrackPalette->setPixmap(pixmap);
+}
+
+void DlgPrefColors::slotHotcuePaletteChanged(const QString& paletteName) {
+    QPixmap preview = drawPalettePreview(paletteName);
+    labelHotcuePalette->setPixmap(preview);
+
+    ColorPalette palette = mixxx::PredefinedColorPalettes::kDefaultTrackColorPalette;
+    foreach (const ColorPalette& pal, mixxx::PredefinedColorPalettes::kPalettes) {
+        if (paletteName == pal.getName()) {
+            palette = pal;
+            break;
+        }
+    }
+
+    comboBoxHotcueDefaultColor->clear();
+
+    QPixmap pixmap(80, 80);
+    QPainter painter(&pixmap);
+    pixmap.fill(Qt::black);
+
+    comboBoxHotcueDefaultColor->addItem(tr("By hotcue number"), -1);
+    for (int i = 0; i < palette.size() && i < 4; ++i) {
+        painter.setPen(mixxx::RgbColor::toQColor(palette.at(i)));
+        painter.setBrush(mixxx::RgbColor::toQColor(palette.at(i)));
+        painter.drawRect(0, i * 20, 80, 20);
+    }
+    comboBoxHotcueDefaultColor->setItemIcon(0, QIcon(pixmap));
+
+    for (int i = 0; i < palette.size(); ++i) {
+        comboBoxHotcueDefaultColor->addItem(tr("Palette") +
+                        QStringLiteral(" ") + QString::number(i + 1),
+                i);
+        pixmap.fill(mixxx::RgbColor::toQColor(palette.at(i)));
+        comboBoxHotcueDefaultColor->setItemIcon(i + 1, QIcon(pixmap));
+    }
+
+    bool autoHotcueColors =
+            m_pConfig->getValue(ConfigKey("[Controls]", "auto_hotcue_colors"), false);
```

Comment:
How about we change the name of this ConfigKey to `hotcue_colors_by_number`? The current name is unclear. I don't think we should worry about breaking configurations considering we haven't released a beta yet and it's easy to change this.

Reply:
In this case, I think we should get entirely rid of it.

Reply:
Then how do you propose to keep track of whether to color hotcues by number or not? I suppose we could use a special value like -1 for HotcueDefaultColorIndex, but I think it's more appropriate to keep a separate boolean ConfigKey.

Reply:
This should be  done after we have considered @Holzhaus plans in a follow up PR.

Reply:
I don't understand what you mean. What would be done in that follow up PR? Are you talking about #2578? I think that PR is made obsolete by this one and we should close #2578 without merging.

Reply:
Ping. @Holzhaus what do you think?

Reply:
The point of #2578 to remove auto_hotcue_colors as an option and always as behavior. Instead, I was planning to make the single default color combobox simply create a new palette on the fly where the hotcue color index is set to 1 for that color and unset it for all other colors. That would make it possible to always use palette.getColorForHotcue() conditionally without using if-else-ladders and checking CO objects. If this PR has been merged i'll look into rebasing #2578 and check if that is indeed a feasible solution.

Reply:
That's similar to what I [proposed](https://github.com/mixxxdj/mixxx/pull/2578#issuecomment-603000399) in #2578. However, I think implementing that would be more trouble than it is worth. What we have here is working and the implementation makes sense. I would just like to rename the `auto_hotcue_colors` ConfigKey to something more descriptive.

Reply:
I'd prefer to switch the name of a ConfigKey now than during beta.

Reply:
ping

Reply:
@daschuer what do you think?

Reply:
The current name is not that bad IMHO
I like to keep it either as it is to not break existing configurations or dispose this parameter entirely and reflect the selected combobox value in one parameter.

Reply:
I do think `auto_hotcue_colors` is not a good name. On its own I have no idea what it means. `hotcue_colors_by_number` would make the code easier to understand. However, I'm not going to insist on blocking this PR over this.